Core/MemMapFunctions.cpp

```diff
@@ -86,21 +72,12 @@ inline void ReadFromHardware(T &var, const u32 address) {
 		// More RAM (remasters, etc.)
 		var = *((const T*)GetPointerUnchecked(address));
 	} else {
-		// In jit, we only flush PC when bIgnoreBadMemAccess is off.
-		if (g_Config.iCpuCore == (int)CPUCore::JIT && g_Config.bIgnoreBadMemAccess) {
-			WARN_LOG(MEMMAP, "ReadFromHardware: Invalid address %08x", address);
-		} else {
-			WARN_LOG(MEMMAP, "ReadFromHardware: Invalid address %08x PC %08x LR %08x", address, currentMIPS->pc, currentMIPS->r[MIPS_REG_RA]);
-		}
 		static bool reported = false;
 		if (!reported) {
 			Reporting::ReportMessage("ReadFromHardware: Invalid address %08x near PC %08x LR %08x", address, currentMIPS->pc, currentMIPS->r[MIPS_REG_RA]);
 			reported = true;
 		}
-		if (!g_Config.bIgnoreBadMemAccess) {
-			Core_EnableStepping(true);
-			host->SetDebugMode(true);
-		}
+		Core_MemoryException(address, currentMIPS->pc, MemoryExceptionType::READ_WORD);
```

Comment:
WORD/etc. depends on T.  Could even be 64 bit.  Though I suppose they're all probably word reads with masking and shifting, really.

-[Unknown]

Reply:
I was too lazy to go all the way and separate the access widths, might do it later.